index.js

```diff
@@ -0,0 +1,9 @@
+const calculateTotal = (items) => items.map(calculatedSubTotals)
+
+const calculatedSubTotals = (items) => {
+  items.total = Number((items.price * items.quantity - (items.price * items.quantity * items.discount)).toFixed(2))
```

Comment:
Map will return a new array which makes calculateTotal pure but this `items.total = ` changes the item passed in so calculatedSubTotals is not pure.